model/lifecycle.go

```diff
@@ -262,23 +262,27 @@ func SetTaskPriority(ctx context.Context, t task.Task, priority int64, caller st
 
 // SetBuildPriority updates the priority field of all tasks associated with the given build id.
 func SetBuildPriority(ctx context.Context, buildId string, priority int64, caller string) error {
-	_, err := task.UpdateAll(
-		bson.M{task.BuildIdKey: buildId},
+	query := bson.M{task.BuildIdKey: buildId}
+	_, err := task.UpdateAll(query,
 		bson.M{"$set": bson.M{task.PriorityKey: priority}},
 	)
 	if err != nil {
 		return errors.Wrapf(err, "setting priority for build '%s'", buildId)
 	}
 
-	// negative priority - these tasks should never run, so unschedule now
+	tasks, err := task.FindAll(db.Query(query))
```

Comment:
Since this logic is duplicated in two places can we move it into a helper function?